src/side_offsets.rs

```diff
@@ -138,3 +138,150 @@ impl<T: Copy + Zero, U> TypedSideOffsets2D<T, U> {
         TypedSideOffsets2D::new(Zero::zero(), Zero::zero(), Zero::zero(), Zero::zero())
     }
 }
+
+/// 3D offsets for a cuboid: front, left, right, back, top, bottom.
+#[derive(EuclidMatrix)]
+#[repr(C)]
+pub struct TypedSideOffsets3D<T, U> {
```

Comment:
Still on the same topic, I propose extracting TypedSideOffsets3D into a separate pull request so that we can land the rest without stalling while figuring out if/how we want to fixate a meaning of "front" and "back", and whether they conflict with the wording of web standards like CSS 3D transforms and WebVR.